### PR TITLE
Add UI action for creating text along line annotation items 

### PR DESCRIPTION
**Pull request by nyalldawson**
And a properties panel for these.

The map tool interaction with the text baseline curve still needs some refinement, but these changes allow users to create their own text-along-line annotation items

Sponsored by North Road, thanks to SLYR

### CODE REVIEW

File: src/gui/annotations/qgsannotationitemwidget_impl.cpp [hunk 1]
```diff
@@ -428,5 +429,161 @@ void QgsAnnotationPointTextItemWidget::mInsertExpressionButton_clicked()
...
+  mPropertiesWidget->setContext( context );
+}
+
+void QgsAnnotationLineTextItemWidget::focusDefaultWidget()
+{
+  mTextEdit->selectAll();
+  mTextEdit->setFocus();
+}
+
+QgsAnnotationLineTextItemWidget::~QgsAnnotationLineTextItemWidget() = default;
```
Comment (nirvn) [inline src/gui/annotations/qgsannotationitemwidget_impl.cpp:534]: Completely up to you here but I prefer destructors next to constructors :)

File: src/ui/annotations/qgsannotationlinetextwidgetbase.ui [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+       <widget class="QLabel" name="label_4">
+        <property name="text">
+         <string>Offset from line</string>
+        </property>
+       </widget>
+      </item>
+     </layout>
+    </widget>
+   </item>
+   <item row="5" column="0">
```
Comment (nirvn) [inline src/ui/annotations/qgsannotationlinetextwidgetbase.ui:119]: That should be row="3".

**Review by nirvn — APPROVED** [review]

File: src/ui/annotations/qgsannotationlinetextwidgetbase.ui (hunk 2 above)
Comment (nyalldawson) [reply src/ui/annotations/qgsannotationlinetextwidgetbase.ui:119]: actually it shouldn't -- QtCreator doesn't store rows/columns in order, and there's a row=3 elsewhere

Comment (nirvn) [reply src/ui/annotations/qgsannotationlinetextwidgetbase.ui:119]: Right, row="4" then? But in any case, there's no unwanted margin in the UI if we skip a row right?

Comment (nyalldawson) [reply src/ui/annotations/qgsannotationlinetextwidgetbase.ui]: I couldn't see one, but I've fixed anyway

**Pull request merged**